### ISSUE TITLE
Rich editor configuration per field

### ISSUE THREAD

**Issue by salvadormrf** [wagtail/wagtail#1284]
There is some things that could be added to rich text:
- add support for different configurations
- needs a way to unregister some plugins globally (there is only hooks for enabling a hallojs plugin)
- make default whitelister configurable, for example disable "a" tag globally.

Example of rich text editor configuration for a field

``` python
# Simple TextArea, where we nee just a few options on editor
class SimpleDbWhitelister(WagtailDbWhitelister):
    element_rules = {
        '[document]': allow_without_attributes,
        'a': attribute_rule({'href': check_url}),
        'p': allow_without_attributes,
        'b': allow_without_attributes,
        'i': allow_without_attributes,
        'u': allow_without_attributes,
    }


class SimpleRichTexArea(WagtailRichTextArea):
    hallo_plugins = {
        'halloheadings': {
            'formatBlocks': []
        },
        'halloformat': {
            'formattings': {
                "bold": True,
                "italic": True,
            },
        },
        'hallolists': {
            "lists": {
                "ordered": False,
                "unordered": False
            }
        },
        'hallowagtaillink': {},
        'hallorequireparagraphs': {},
        'hallocleanhtml': {
            'format': False,
            'removeTags': ['span', 'div', 'table', 'strong'],
            'allowedTags': ['a', 'p', 'i', 'b', 'u'],
            'removeAttrs': ['class', 'style', 'id'],
            'allowedAttributes': [
                ["a", ['href', 'target', 'data-id', 'data-linktype']]
            ]
        }
    }

    def render_js_init(self, id_, name, value):
        return "makeRichTextEditable({0}, {1});".format(
            json.dumps(id_),
            json.dumps(self.hallo_plugins)
        )

    def value_from_datadict(self, data, files, name):
        original_value = super(SimpleRichTexArea, self).value_from_datadict(data, files, name)
        if original_value is None:
            return None
        return SimpleDbWhitelister.clean(original_value)
```

Some changes on page-editor.js

``` javascript
function makeRichTextEditable(id, plugins) {
   var instanceHalloPlugins = plugins || halloPlugins;
....
}
```

Globally disable a hallo plugin on editor

``` javascript
function unRegisterHalloPlugin(name) {
    if (!$.inArray(name, disabledHalloPlugins)) {
        disabledHalloPlugins.push(name);
    }
    if ($.inArray(name, halloPlugins)) {
        delete halloPlugins[name];
    }
}
```

**Comment by mattots** [comment]
Yes this is something I'd really like to be able to do too. At the very least, to be able to configure the global list of plugins. (For example, I'd like to remove the ability to add images and embeds to rich text areas, as I'm going to be providing separate fields inside StreamField blocks for adding these so I can better control their display.)

**Comment by SalahAdDin** [comment]
Wow, awesome, is the better wat for customize the richtextfield editor for the content. Thank you.

**Comment by drwlrsn** [comment]
I mentioned wanting something like this in the Google Group. Now with `StreamField`s, being able to provide a stripped down editor for `RichTextField`s is important so editors will use the fields provided and not use a `RichTextField` for content that should be in a provided field.

**Comment by sondrele** [comment]
+1

The ability to disable things like images and embeds would be great in combination with StreamField

**Comment by SalahAdDin** [comment]
:+1:

**Comment by hacksaw6** [comment]
+1

**Comment by nek4life** [comment]
+1 We need this functionality in order to provide a stripped down editor functionality for things like links in photo captions where we do not want the author to be able to add anything else but an anchor.

**Comment by wiebl** [comment]
+1

**Comment by noahmanger** [comment]
+1. Has anyone found good ways to deal with this at the moment?

**Comment by gasman** [comment]
I believe #2697 (now in master) should address this. Haven't had chance to try it out myself, and it still needs docs, but if anyone's up for trying it out and reporting back, that would be much appreciated!

**Comment by gasman** [comment]
Now implemented in #3736.

(To correct myself above: #2697 only addresses this in a very basic way - you'd need to set up a configuration in `WAGTAILADMIN_RICH_TEXT_EDITORS` for each combination of features - and only for editors that recognise configuration options passed there, which hallo.js didn't until I implemented that in #3736)

**Comment by gasman** [comment]
Completed in #3736
